source/auth/auth.rst

```diff
@@ -453,10 +569,62 @@ Implementation
 Test Plan
 =========
 
-Tests have been defined in the associated files:
+Connection string tests have been defined in the associated files:
 
 * `Connection String <tests/connection-string.json>`_.
 
+---------------------------------------
+SCRAM-SHA-256 and mechanism negotiation
+---------------------------------------
+
+Testing SCRAM-SHA-256 requires server version 3.7.3 or later with
+``featureCompatibilityVersion`` of "4.0" or later.
+
+Drivers that allow specifying auth parameters in code as well as via
+connection string should test both for the test cases described below.
+
+Step 1
+------
+
+Create three test users, one with only SHA-1, one with only SHA-256 and one
+with both.  For example::
+
+    db.createUser({user: 'sha1', pwd: 'sha256', roles: ['root'], mechanisms: ['SCRAM-SHA-1']})
+    db.createUser({user: 'sha256', pwd: 'sha256', roles: ['root'], mechanisms: ['SCRAM-SHA-256']})
+    db.createUser({user: 'both', pwd: 'both', roles: ['root'], mechanisms: ['SCRAM-SHA-1', 'SCRAM-SHA-256']})
+
+Optional, but strongly encouraged: Create a SCRAM-SHA-256 user with a username
+and password that are modified by SASLprep, for example ``"\u2168"`` (ROMAN
+NUMERAL NINE), which is transformed by SASLprep to ``IV``.  Note: when calling
+the ``createUser`` command, the username must be prepped or the command will
+fail.  The password need not be prepped.
+
+Step 2
+------
+
+For each test user, verify that you can connect and run a command requiring
+authentication for the following cases:
+
+- Explicitly specifying each mechanism the user supports.
+- Specifying no mechanism and relying on mechanism negotiation.
+
+For the example users above, the ``dbstats`` command could be used as a test
+command.
+
+For a test user supporting both SCRAM-SHA-1 and SCRAM-SHA-256, drivers should
+verify that negotation selects SCRAM-SHA-256.  This may require monkey
+patching, manual log analysis, etc.
+
+Step 3
+------
+
+For test users that support only one mechanism, verify that explictly specifying
+the other mechanism fails.
+
+For a non-existent username, verify that not specifying a mechanism when
```

Comment:
The requirement should be to ensure that auth fails with the same error type as providing the correct user but the wrong password. Some drivers, like PyMongo, don't have a specific exception type for authentication failure.

Reply:
Done